execution/executor/src/block_processor.rs

```diff
@@ -43,23 +40,32 @@ enum Mode {
     Syncing,
 }
 
+fn error_when_syncing(id: HashValue) -> Error {
+    let message = format!("Syncing. Unable to serve request for block {:x}.", id);
+    warn!("{}", message);
+    format_err!("{}", message)
+}
+
 pub(crate) struct BlockProcessor<V> {
     /// Where the processor receives commands.
     command_receiver: mpsc::Receiver<Command>,
 
     /// The timestamp of the last committed ledger info.
     committed_timestamp_usecs: u64,
 
-    committed_trees: ExecutedTrees,
+    committed_trees: Arc<Mutex<ExecutedTrees>>,
 
     /// The main block tree data structure that holds all the uncommitted blocks in memory.
-    block_tree: BlockTree<TransactionBlock>,
+    blocks_to_execute: VecDeque<(
+        ExecutableBlock,
+        oneshot::Sender<Result<(ProcessedVMOutput, StateComputeResult)>>,
+    )>,
```

Comment:
comments need update. also i'm not sure why we need to queue these blocks here. probably can get rid of this in another diff.